### PR TITLE
Format multiline strings as block arguments.

### LINKED ISSUE #570: scheduled_test descriptor pattern is indented very strangely
The formatter doesn't play nicely at all with the following pattern from `scheduled_test`'s `descriptor` library:

```dart
void main() {
  d.file("test.dart", """
    import 'package:test/test.dart';
    
    void main() {
      test("failure", () => throw new TestFailure("oh no"));
    }
  """).create();
}
```

is formatted as:

```dart
void main() {
  d
      .file(
          "test.dart",
          """
    import 'package:test/test.dart';
    
    void main() {
      test("failure", () => throw new TestFailure("oh no"));
    }
  """)
      .create();
}
```

This is definitely a lot harder to read.

### PR DESCRIPTION
**Pull request by nex3**
Closes #570

Note that I'm not wedded to the nomenclature here at all.

### CODE REVIEW

File: lib/src/source_visitor.dart [hunk 1]
```diff
@@ -68,20 +68,19 @@ class SourceVisitor extends ThrowingAstVisitor {
...
+  /// also tracks the chunk before the argument that determines whether or not
+  /// the block body is indented like an expression or a statement.
   ///
-  /// Before a collection literal argument is visited, [ArgumentSublist] binds
-  /// itself to the left bracket token of each collection literal it controls.
-  /// When we later visit that literal, we use the token to find that
-  /// association.
-  final Map<Token, ArgumentSublist> _collectionArgumentLists = {};
+  /// Before a block argument is visited, [ArgumentSublist] binds itself to the
+  /// left bracket token of each collection literal it controls. When we later
```
Comment (munificent) [inline lib/src/source_visitor.dart]: "left bracket token of each collection literal it controls" → "beginning token of each block it controls".

File: test/splitting/strings.stmt [hunk 2]
```diff
@@ -53,4 +53,24 @@ someMethod("""first line does not fits here
...
+>>> format multiline a block with a trailing argument
+someMethod("""
+  some
+  text
+""", "foo");
+<<<
+someMethod("""
+  some
+  text
+""", "foo");
```
Comment (munificent) [inline test/splitting/strings.stmt]: Can you test:

* Some negative cases, like a multiline string with non-block arguments before and after.
* A sequence of multiline strings in an argument list.
* A mixture of multiline strings and other block arguments like collection literals.

File: lib/src/source_visitor.dart (hunk 1 above)
Comment (nex3) [inline lib/src/source_visitor.dart]: Done.

File: test/splitting/strings.stmt (hunk 2 above)
Comment (nex3) [inline test/splitting/strings.stmt]: > Some negative cases, like a multiline string with non-block arguments before and after.

Did one. I'm not sure what other negative cases are meaningfully different, though.

> A sequence of multiline strings in an argument list.

Done.

> A mixture of multiline strings and other block arguments like collection literals.

Done.

**Pull request merged**